src/response.ts

```diff
@@ -0,0 +1,31 @@
+import { FastifyReply } from 'fastify';
+
+// response handling function
+
+export function handleServerResponse(
+  reply: FastifyReply,
+  message: string,
+  code: number,
+  data: unknown
```

Comment:
Why not join both into one and keeping data and error key as optional parameter

Reply:
# if i combine both functions i will have to add "fastify reply" to response handlers as well .

## the code will look like this
```
  const responseHandler = (reply,message,code,data=null,error=null) =>{
        some function
    }

```
but when we call the function in a error handling
```
   responseHandler(reply,"some message",404,null,{error:errorobj})

```
 it kinda feels inconvenient to send "fastify reply" and null unwantedly

Reply:
Lastly. If your still feeling why reply.send moving it inside is a bad idea. I would request you to write a testcase for this file. Then you will realize why this will bite you back

Reply:
for success reponse

```
export const rootHandler = async (_req: FastifyRequest, reply: FastifyReply) => {
  const res = renderResponse(true, 'something went wrong', { data: 'data' });
  reply.status(201).send(res);
};
```
for error
  here error code is 404 by default
```
export const rootHandler = async (_req: FastifyRequest, reply: FastifyReply) => {
  const res = renderResponse(true, 'something went wrong', { data: 'data' },400);
  reply.status(201).send(res);
};
```
didn't  added enum